www/shared/components/cookie-banner/CookieBanner.test.js

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react';
+
+const mockCookieConsents = [];
+const mockAdd = jest.fn().mockImplementation((cookie) => mockCookieConsents.push(cookie));
+
+jest.doMock('@enzsft/react-cookie-consents',
+    () => ({
+        useCookieConsents: () => ({
+            get: () => mockCookieConsents,
+            add: mockAdd,
+        }),
+    }),
+);
+
+it('should render banner when cookie consent was not accepted', () => {
+    const CookieBanner = require('./CookieBanner').default;
+
+    const { container } = render(
+        <CookieBanner />,
+    );
+
+    expect(container).toHaveTextContent('cookies');
+});
+
+it('should add analytics when accept button is clicked', () => {
+    const CookieBanner = require('./CookieBanner').default;
+
+    const { getByText } = render(
+        <CookieBanner />,
+    );
+
+    const buttonNode = getByText('Accept');
+
+    fireEvent.click(buttonNode);
+
+    expect(mockAdd).toHaveBeenCalled();
+    expect(mockCookieConsents).toHaveLength(1);
```

Comment:
True, thanks!  🙏

Reply:
I replaced this assertion with `   expect(mockCookieConsents).toEqual(expect.arrayContaining(['analytics']));` because that's the cookie we want to check if it was added.

Reply:
I'd say we should assert `mockAdd` is called with the appropriate cookie instead of asserting the contents of the array.

Reply:
Yep, maybe that's better and it's also more readable 😄

Reply:
It is, but it is also a matter of correctness. We want to make sure the component calls `add` as expected.